consensus/state_sync/state_sync.go

```diff
@@ -0,0 +1,102 @@
+package state_sync
+
+import (
+	typesCons "github.com/pokt-network/pocket/consensus/types"
+	cryptoPocket "github.com/pokt-network/pocket/shared/crypto"
+	"github.com/pokt-network/pocket/shared/modules"
+)
+
+type StateSyncModule interface {
+	modules.Module
```

Comment:
We have 3 `Run` functions here, each of which can be started/stopped independently; with the exception of synched and sync mode running concurrently.

I was envisioning that each of them would be an independent module, so something like `ServerMode` can be started and stopped separately.

Reply:
I can see that as a real possibility

Reply:
![ea6955dd-4506-473d-9abb-806dd0464797_text](https://user-images.githubusercontent.com/55156619/204351516-1e409841-a203-448a-8e0e-492bc28feefd.gif) - Dwight Schrute - Andrew Nguyen